Escape quote in license string

locnide opened the pull request:
The license string in the generated nix package file was broken when the license name contained quote.

It has been fixed by escaping it as suggested.

closes https://github.com/lopsided98/nix-ros-overlay/issues/356

locnide committed:
nix: fix: escape special characters in license

The license string in the generated nix package file was broken when the
license name contained quote.

It has been fixed. Backslash and the sequence "dollar opening bracket"
have been escaped too.

hacker1024 commented:
I know this is unlikely to occur in practice, but we should probably implement a similar check for `${` and `\` (if `\"` is already included, it would get converted to `\\"`, which is not helpful).

locnide commented:
> I know this is unlikely to occur in practice, but we should probably implement a similar check for `${` and `\` (if `\"` is already included, it would get converted to `\\"`, which is not helpful).

Thank you for your answer, I added tests for those two cases and rebased.
